pgpainless-core/src/main/java/org/pgpainless/key/modification/secretkeyring/SecretKeyRingEditor.java

```diff
@@ -286,6 +287,49 @@ public SecretKeyRingEditorInterface revokeSubKey(long subKeyId,
         return this;
     }
 
+    @Override
+    public SecretKeyRingEditorInterface setExpirationDate(Date expiration,
+                                                          SecretKeyRingProtector secretKeyRingProtector)
+            throws PGPException {
+        List<PGPSecretKey> secretKeyList = new ArrayList<>();
+        Iterator<PGPSecretKey> secretKeyIterator = secretKeyRing.getSecretKeys();
+
+        if (secretKeyIterator.hasNext()) {
```

Comment:
I'd invert this if statement and throw an exception if the iterator has no next entry.

Reply:
Hmm... what if the iterator has multiple entries? Maybe we need to document that we'd process only the first secret key :thinking:

Reply:
If the iterator has multiple entries those are the subkeys, right?
Can subkeys expire independantly from the primary key?

Reply:
Yeah, definitely subkeys can expire independently (see my key for example). There are some interesting uses for that, for example one can emulate forward secrecy by creating multiple encryption subkeys where each subkey is valid for a week.

But this is definitely less important than the primary key expiry. I was thinking on renaming this to `setPrimaryKeyExpiration` but a more generic way would be to pass a KeyID and do the appropriate logic depending on if we have primary key KeyID or a subkey.

(Currently I'm planning to add keyId parameter, if it's primary key, do what do we have now, if not, throw, later we can extend that) :)